adapters/repos/db/shard_init_vector.go

```diff
@@ -255,7 +256,12 @@ func (s *Shard) initVectorIndex(ctx context.Context,
 				WriteSegmentInfoIntoFileName: s.index.Config.SegmentInfoIntoFileNameEnabled,
 				WriteMetadataFilesEnabled:    s.index.Config.WriteMetadataFilesEnabled,
 			},
-			TombstoneCallbacks: s.cycleCallbacks.vectorTombstoneCleanupCallbacks,
+			SplitWorkers:              w,
+			ReassignWorkers:           w,
+			InternalPostingCandidates: spfreshent.DefaultInternalPostingCandidates,
```

Comment:
Relying on the "user"  to set external default values for an internal API is not ideal.
We should instead do something more idiomatic:
- As a caller, only pass values you want to override
- In the SPFresh constructor, make sure to validate the values given by the caller and set the default values if they were not passed